### PR TITLE
Button select group

### PR DESCRIPTION
**Pull request by bwalker8038**
## Description
This `pr` adds the `<ButtonSelectGroup />` component to the design-system.

## Examples

With mouse navigation

select with default

With Voiceover/keyboard navigation

voiceover-button-group

## Changes

- Added the `<ButtonSelectGroup/>` component
- Added the `<OptionButton />` sub-component
- Added tests
- Added supporting `aria` tags to the `<Button />` component
- Added documentation

### CODE REVIEW

File: src/components/Button/Button.js [hunk 1]
```diff
@@ -75,16 +77,23 @@ function PrivateButton({
   if (isSelected) classNames.push('isSelected')
   if (arrowIcon) classNames.push('arrowIcon')
 
+  let checked
+  if (isSelected) {
+    classNames.push('isSelected')
+    checked = { 'aria-checked': isSelected }
```
Comment (kenning) [inline src/components/Button/Button.js]: 😍

File: src/components/ButtonSelectGroup/ButtonSelectGroup.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
+import React, { useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import uuidv4 from 'uuid/v4'
+import { OptionButton } from './OptionButton'
+import { Body } from '../Type/Body'
+import { COLORS } from '../Colors'
+import './ButtonGroup.scss'
+
+/**
+ **/
```
Comment (kenning) [inline src/components/ButtonSelectGroup/ButtonSelectGroup.js]: Could delete these lines at some point, not blocking

File: src/components/ButtonSelectGroup/ButtonSelectGroup.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+ *
+ * @example ```
+ * <ButtonSelectGroup
+ *  defaultValue="excellent"
+ *  label="Health"
+ *  onSelect={({ value }) => console.log(value)}
+ * >
+ *   <ButtonSelectGroup.Option value="average">Average</ButtonSelectGroup.Option>
+ *   <ButtonSelectGroup.Option value="great">Greate</ButtonSelectGroup.Option>
+ * </ButtonSelectGroup>
```
Comment (zach-ethos) [inline src/components/ButtonSelectGroup/ButtonSelectGroup.js]: Love this interface.^

File: src/components/ButtonSelectGroup/ButtonSelectGroup.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  label="Health"
+  onSelect={({ value }) => console.log(value)}
+>
+  <ButtonSelectGroup.Option onClick={() => console.log('i was clicked!')} value="average">Average</ButtonSelectGroup.Option>
+  <ButtonSelectGroup.Option value="great">Great</ButtonSelectGroup.Option>
+  <ButtonSelectGroup.Option value="excellent">
+    Exellent
+  </ButtonSelectGroup.Option>
+</ButtonSelectGroup>
+```
```
Comment (zach-ethos) [inline src/components/ButtonSelectGroup/ButtonSelectGroup.md]: A+++

**Comment by zach-ethos** [comment]
> You forgot to add the component as an export of the design system in src/components/index.js, but otherwise looks great.

Still catching up, but is this still necessary now that we're bundling (instead of just exporting/importing src)? cc @kenning @bwalker8038

**Pull request merged**